File: proplet-rs/src/monitoring/profiles.rs
use crate::types::MonitoringProfile;
use std::time::Duration;

impl MonitoringProfile {
    pub fn minimal() -> Self {
        Self {
            enabled: true,
            interval: Duration::from_secs(60),
            collect_cpu: true,
            collect_memory: true,
            collect_disk_io: false,
            collect_network_io: false,
            collect_threads: false,
            collect_file_descriptors: false,
            export_to_mqtt: false,
            retain_history: false,
            history_size: 0,
        }
    }

    pub fn standard() -> Self {
        Self {
            enabled: true,
            interval: Duration::from_secs(10),
            collect_cpu: true,
            collect_memory: true,
            collect_disk_io: true,
            collect_network_io: true,
            collect_threads: true,
            collect_file_descriptors: true,
            export_to_mqtt: true,
            retain_history: true,
            history_size: 100,
        }
    }

    pub fn intensive() -> Self {
        Self {
            enabled: true,
            interval: Duration::from_secs(1),
            collect_cpu: true,
            collect_memory: true,
            collect_disk_io: true,
            collect_network_io: true,
            collect_threads: true,
            collect_file_descriptors: true,
            export_to_mqtt: true,
            retain_history: true,
            history_size: 1000,
        }
    }

    pub fn batch_processing() -> Self {
        Self {
            enabled: true,
            interval: Duration::from_secs(30),
            collect_cpu: true,
            collect_memory: true,
            collect_disk_io: true,
            collect_network_io: false,
            collect_threads: false,
            collect_file_descriptors: false,
            export_to_mqtt: true,
            retain_history: true,
            history_size: 200,
        }
    }

    pub fn real_time_api() -> Self {
        Self {
            enabled: true,
            interval: Duration::from_secs(5),
            collect_cpu: true,
            collect_memory: true,
            collect_disk_io: false,
            collect_network_io: true,
            collect_threads: true,
            collect_file_descriptors: true,
            export_to_mqtt: true,
            retain_history: true,
            history_size: 500,
        }
    }

    pub fn long_running_daemon() -> Self {
        Self {
            enabled: true,
            interval: Duration::from_secs(120),
            collect_cpu: true,
            collect_memory: true,
            collect_disk_io: true,
            collect_network_io: true,
            collect_threads: true,
            collect_file_descriptors: true,
            export_to_mqtt: true,
            retain_history: true,
            history_size: 500,
        }
    }

    pub fn disabled() -> Self {
        Self {
            enabled: false,
            interval: Duration::from_secs(60),
            collect_cpu: false,
            collect_memory: false,
            collect_disk_io: false,
            collect_network_io: false,
            collect_threads: false,
            collect_file_descriptors: false,
            export_to_mqtt: false,
            retain_history: false,
            history_size: 0,
        }
    }
}

File: pkg/proplet/monitoring/profiles.go
package monitoring

import "time"

type MonitoringProfile struct {
	Enabled                bool          `json:"enabled"`
	Interval               time.Duration `json:"interval"`
	CollectCPU             bool          `json:"collect_cpu"`
	CollectMemory          bool          `json:"collect_memory"`
	CollectDiskIO          bool          `json:"collect_disk_io"`
	CollectNetworkIO       bool          `json:"collect_network_io"`
	CollectThreads         bool          `json:"collect_threads"`
	CollectFileDescriptors bool          `json:"collect_file_descriptors"`
	ExportToMQTT           bool          `json:"export_to_mqtt"`
	RetainHistory          bool          `json:"retain_history"`
	HistorySize            int           `json:"history_size"`
}

func StandardProfile() MonitoringProfile {
	return MonitoringProfile{
		Enabled:                true,
		Interval:               10 * time.Second,
		CollectCPU:             true,
		CollectMemory:          true,
		CollectDiskIO:          true,
		CollectNetworkIO:       true,
		CollectThreads:         true,
		CollectFileDescriptors: true,
		ExportToMQTT:           true,
		RetainHistory:          true,
		HistorySize:            100,
	}
}

func MinimalProfile() MonitoringProfile {
	return MonitoringProfile{
		Enabled:                true,
		Interval:               60 * time.Second,
		CollectCPU:             true,
		CollectMemory:          true,
		CollectDiskIO:          false,
		CollectNetworkIO:       false,
		CollectThreads:         false,
		CollectFileDescriptors: false,
		ExportToMQTT:           false,
		RetainHistory:          false,
		HistorySize:            0,
	}
}

func IntensiveProfile() MonitoringProfile {
	return MonitoringProfile{
		Enabled:                true,
		Interval:               1 * time.Second,
		CollectCPU:             true,
		CollectMemory:          true,
		CollectDiskIO:          true,
		CollectNetworkIO:       true,
		CollectThreads:         true,
		CollectFileDescriptors: true,
		ExportToMQTT:           true,
		RetainHistory:          true,
		HistorySize:            1000,
	}
}

func BatchProcessingProfile() MonitoringProfile {
	return MonitoringProfile{
		Enabled:                true,
		Interval:               30 * time.Second,
		CollectCPU:             true,
		CollectMemory:          true,
		CollectDiskIO:          true,
		CollectNetworkIO:       false,
		CollectThreads:         false,
		CollectFileDescriptors: false,
		ExportToMQTT:           true,
		RetainHistory:          true,
		HistorySize:            200,
	}
}

func RealTimeAPIProfile() MonitoringProfile {
	return MonitoringProfile{
		Enabled:                true,
		Interval:               5 * time.Second,
		CollectCPU:             true,
		CollectMemory:          true,
		CollectDiskIO:          false,
		CollectNetworkIO:       true,
		CollectThreads:         true,
		CollectFileDescriptors: true,
		ExportToMQTT:           true,
		RetainHistory:          true,
		HistorySize:            500,
	}
}

func LongRunningDaemonProfile() MonitoringProfile {
	return MonitoringProfile{
		Enabled:                true,
		Interval:               120 * time.Second,
		CollectCPU:             true,
		CollectMemory:          true,
		CollectDiskIO:          true,
		CollectNetworkIO:       true,
		CollectThreads:         true,
		CollectFileDescriptors: true,
		ExportToMQTT:           true,
		RetainHistory:          true,
		HistorySize:            500,
	}
}

func DisabledProfile() MonitoringProfile {
	return MonitoringProfile{
		Enabled:                false,
		Interval:               60 * time.Second,
		CollectCPU:             false,
		CollectMemory:          false,
		CollectDiskIO:          false,
		CollectNetworkIO:       false,
		CollectThreads:         false,
		CollectFileDescriptors: false,
		ExportToMQTT:           false,
		RetainHistory:          false,
		HistorySize:            0,
	}
}
